File: langgraph-quiz-nodejs/agents/_lib/state.ts
import { StateSchema } from "@langchain/langgraph";
import { z } from "zod";

export const MAX_ATTEMPTS = 2;

export const QuizState = new StateSchema({
  current_question: z.string(),
  options: z.array(z.string()),
  correct_option: z.string(),

  user_answer: z.string(),
  is_correct: z.boolean(),

  is_first_attempt: z.boolean(),
  hint_given: z.boolean(),
  last_feedback: z.string(),

  question_number: z.number(),
  total_questions: z.number(),
  score: z.number(),
  total_attempts: z.number(),

  language: z.string(),
});

export type QuizStateType = typeof QuizState.State;

File: langgraph-quiz-python/agents/_lib/state.py
from typing_extensions import TypedDict

MAX_ATTEMPTS = 2


class QuizState(TypedDict, total=False):
    current_question: str
    options: list[str]
    correct_option: str

    user_answer: str
    is_correct: bool

    is_first_attempt: bool
    hint_given: bool
    last_feedback: str

    question_number: int
    total_questions: int
    score: int
    total_attempts: int

    question_history: list[dict]

    language: str
